gocourse05/security/security.go

```diff
@@ -0,0 +1,111 @@
+package security
+
+import "fmt"
+
+type Range float32
+type Angle float32
+
+type AnimalRecord struct {
+	AnimalType
+	AnimalAction
+}
+
+func (r AnimalRecord) Print() {
+	fmt.Printf("Animal %s is doing %s \n", r.AnimalType, r.AnimalAction)
+}
+
+func NewAnimalRecord(t AnimalType, a AnimalAction) *AnimalRecord {
+	return &AnimalRecord{
+		AnimalType:   t,
+		AnimalAction: a,
+	}
+}
+
+type AnimalType string
+type AnimalAction string
+
+const (
+	Bear  AnimalType = "Bear"
+	Tiger AnimalType = "Tiger"
+	Rat   AnimalType = "Rat"
+
+	Sleep   AnimalAction = "Sleep"
+	Eat     AnimalAction = "Eat"
+	Walking AnimalAction = "Walking"
+)
+
+type Condition struct {
+	IsNight      bool
+	DistToAnimal Range
+	Battery      int
+}
+
+func NewConditions(isNight bool, dst Range, battery int) *Condition {
```

Comment:
Зайвий `s` префікс, здається. Також не дуже добре виглядає, коли одне слово в рядку скорочене, а всі інші - ні. Пропоную так:
```suggestion
func NewCondition(isNight bool, distToAnimal Range, battery int) *Condition {
```